src/pdb/mod.rs

```diff
@@ -641,22 +657,23 @@ impl RowGroup {
             })
     }
 
-    fn row_offsets_writer<Writer>(
-        row_presence_flags: u16,
-    ) -> impl FnOnce(&[u16; 16], &mut Writer, Endian, ()) -> BinResult<()>
+    fn write_row_offsets<Writer>(
+        row_offsets: &[u16; 16],
+        writer: &mut Writer,
+        endian: Endian,
+        (row_presence_flags,): (u16,),
+    ) -> BinResult<()>
     where
         Writer: Write + Seek,
     {
-        move |row_offsets, writer, endian, ()| {
-            let skip = row_presence_flags.leading_zeros() as usize;
-            writer.seek(SeekFrom::Current(
-                (skip * std::mem::size_of::<u16>()) as i64,
-            ))?;
-            for offset in row_offsets.iter().skip(skip) {
-                offset.write_options(writer, endian, ())?;
-            }
-            Ok(())
+        let skip = usize::try_from(row_presence_flags.leading_zeros()).unwrap();
+        writer.seek(SeekFrom::Current(
+            (skip * std::mem::size_of::<u16>()).try_into().unwrap(),
+        ))?;
```

Comment:
sorry, for the NIT, but lets avoid all the `try_into()`. `unwrap()` just because of the unnecessary `usize`. In this specific case, the narrowing cast is fine IMO.
```suggestion
        const U16_byte: u32 = std::mem::size_of::<u16>() as u32;
        let skip = row_presence_flags.leading_zeros();
        writer.seek(SeekFrom::Current((skip * U16_byte).into()))?;
```